### PR TITLE
[android] 📅 Update @react-native-community/datetimepicker to 2.2.2

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Part of #7006 

# How

- Used `et update-vendored-module` to update `@react-native-community/datetimepicker`.
- Updated version of this module in `native-component-list` and updated `DateTimePickerScreen` with the new example app.
- Added changelog entry.

# Test Plan

Tested on both iOS and Android in `native-component-list`. One thing that I noticed is that the returned dates and times are sometimes different between platforms (due to timezone?) but looks like this issue was also happening in the past 🤷‍♂

### CODE REVIEW

File: apps/native-component-list/src/screens/DateTimePickerScreen.tsx [hunk 1]
```diff
@@ -1,96 +1,119 @@
...
-import { StyleSheet, View, Text, Button, Platform } from 'react-native';
+import React, { useState } from 'react';
+import moment from 'moment';
 
 // This example is a copy from https://github.com/react-native-community/react-native-datetimepicker/tree/master/example
 // Please try to keep it up to date when updating @react-native-community/datetimepicker package :)
 
-type DateTimeMode = 'date' | 'datetime' | 'time';
+const DateTimePickerScreen = () => {
+  const [date, setDate] = useState(new Date(1598051730000));
```
Comment (sjchmiela) [inline apps/native-component-list/src/screens/DateTimePickerScreen.tsx]: is it your birthday?

File: yarn.lock [hunk 2]
```diff
@@ -2735,7 +2735,7 @@
     slash "^3.0.0"
     xmldoc "^1.1.2"
 
-"@react-native-community/cli-platform-ios@3.0.0", "@react-native-community/cli-platform-ios@^3.0.0-alpha.1", "@react-native-community/cli-platform-ios@^3.0.0-alpha.7":
+"@react-native-community/cli-platform-ios@^3.0.0-alpha.1", "@react-native-community/cli-platform-ios@^3.0.0-alpha.7":
```
Comment (sjchmiela) [inline yarn.lock]: This shouldn't be here right?

**Review by sjchmiela — APPROVED** [review]
LGTM

File: apps/native-component-list/src/screens/DateTimePickerScreen.tsx (hunk 1 above)
Comment (tsapeta) [reply apps/native-component-list/src/screens/DateTimePickerScreen.tsx]: Hah, I wish to be born on 21st August 2020 😉 
This example is mostly just a copy from:
https://github.com/react-native-community/react-native-datetimepicker/blob/e6d2980230b1e27fa8aed989ccbb79897e6ef5eb/example/App.js#L16-L18

**Pull request merged**